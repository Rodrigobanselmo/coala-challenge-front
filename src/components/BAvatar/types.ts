import { IUser } from "@/core/interfaces/IUser";

export interface IAvatarProps {
  name: string;
  photoUrl?: string | null;
}
