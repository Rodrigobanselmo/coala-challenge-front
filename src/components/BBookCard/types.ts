import { IBook } from "@/core/interfaces/IBook";
import { CircularProgressProps } from "@mui/material/CircularProgress";
import { IconButtonProps } from "@mui/material/IconButton";

export type BBookCardProps = {
  book: IBook;
  buttonLabel?: string;
  onClickButton?: (book: IBook) => Promise<void>;
};
