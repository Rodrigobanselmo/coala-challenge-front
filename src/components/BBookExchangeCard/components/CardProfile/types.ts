import { IUser } from "@/core/interfaces/IUser";

export interface IProfileProps {
  user?: IUser | null;
}
