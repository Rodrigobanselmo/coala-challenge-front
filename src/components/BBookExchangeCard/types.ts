import { IBook } from "@/core/interfaces/IBook";
import { IBookExchange } from "@/core/interfaces/IBookExchange";

export type BBookCardProps = {
  bookExchange: IBookExchange;
  buttonLabel?: string;
  onClickButton?: (book: IBookExchange) => Promise<void>;
};
