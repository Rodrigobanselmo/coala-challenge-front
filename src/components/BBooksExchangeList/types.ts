import { IBookExchange } from "@/core/interfaces/IBookExchange";
import { InfiniteData } from "@tanstack/react-query";
import { ReactNode } from "react";

export interface BBooksExchangeProps {
  data: InfiniteData<IBookExchange[]>;
  searchInputValue: string;
  onSearchInputChange: (value: string) => void;
  fetchNextPage: () => void;
  hasNextPage?: boolean;
  isFetchingNextPage: boolean;
  isLoading: boolean;
  loadMore?: boolean;
  buttonLabel?: string;
  onClickCardButton?: (book: IBookExchange) => Promise<void>;
  placeholder?: string;
  title: string;
  emptyBookComponent?: ReactNode;
  emptyBookLable?: string;
}
