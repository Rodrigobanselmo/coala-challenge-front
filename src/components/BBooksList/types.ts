import { IBook } from "@/core/interfaces/IBook";
import { IPaginationResult } from "@/core/interfaces/IPaginationResult";
import { InfiniteData } from "@tanstack/react-query";
import { ElementType, ReactNode } from "react";

export interface BBooksListProps {
  data: InfiniteData<IBook[]>;
  searchInputValue: string;
  onSearchInputChange: (value: string) => void;
  fetchNextPage: () => void;
  hasNextPage?: boolean;
  isFetchingNextPage: boolean;
  isLoading: boolean;
  loadMore?: boolean;
  buttonLabel?: string;
  onClickCardButton?: (book: IBook) => Promise<void>;
  placeholder?: string;
  emptyBookComponent?: ReactNode;
  emptyBookLable?: string;
  title: string;
}
