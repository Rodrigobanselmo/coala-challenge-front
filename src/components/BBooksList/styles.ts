import { Box, styled } from "@mui/material";

export const StyledBoxContainerList = styled(Box)`
  overflow: scroll;
  padding-bottom: 100px;
  width: 100%;
  flex: 1;
  height: calc(100% - 120px);
  display: flex;
  flex-direction: column;
  align-items: center;
`;
