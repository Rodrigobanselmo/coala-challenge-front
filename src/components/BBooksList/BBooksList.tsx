import React, { FC, Fragment } from "react";

import { Button, LinearProgress } from "@mui/material";
import { BBookCard } from "../BBookCard";
import { StyledGridContainer } from "../BBookCard/styles";
import { BEmptyBlock } from "../BEmptyBlock";
import { BButtonLoadMore } from "../BButtonLoadMore";
import { StyledBoxContainerList } from "./styles";
import { BBooksListProps } from "./types";
import { HeaderList } from "../BHeaderList";

export const BBooksList: FC<BBooksListProps> = ({
  data,
  fetchNextPage,
  isFetchingNextPage,
  isLoading,
  onSearchInputChange,
  searchInputValue,
  buttonLabel,
  onClickCardButton,
  loadMore = true,
  placeholder = "pesquisar...",
  emptyBookComponent,
  emptyBookLable = "Nenhum livro encontrado",
  title,
}: BBooksListProps) => {
  const inputRef = React.useRef<HTMLInputElement>(null);
  const noMoreBooks = !!data?.pages?.[data?.pages?.length - 1]?.length;
  const emptyBooks = !data?.pages?.[0]?.length;

  const onFocuInput = () => {
    inputRef.current?.focus();
  };

  return (
    <>
      <HeaderList
        title={title}
        onSearchInputChange={onSearchInputChange}
        searchInputValue={searchInputValue}
        placeholder={placeholder}
        inputRef={inputRef}
      />
      {(isLoading || isFetchingNextPage) && <LinearProgress />}
      <StyledBoxContainerList>
        {!isLoading && (
          <>
            <StyledGridContainer>
              {data.pages.map((books, i) => (
                <Fragment key={i}>
                  {books.map((book) => (
                    <BBookCard
                      onClickButton={onClickCardButton}
                      book={book}
                      key={book.id || book.googleId}
                      buttonLabel={buttonLabel}
                    />
                  ))}
                </Fragment>
              ))}
            </StyledGridContainer>

            {loadMore && !emptyBooks && (
              <BButtonLoadMore
                canLoadMore={!noMoreBooks}
                fetchNextPage={fetchNextPage}
                isFetchingNextPage={isFetchingNextPage}
              />
            )}

            {emptyBooks && (
              <>
                {emptyBookComponent ? (
                  emptyBookComponent
                ) : (
                  <BEmptyBlock
                    text={emptyBookLable}
                    endComponent={
                      <Button variant="outlined" onClick={() => onFocuInput()}>
                        Adicionar livro
                      </Button>
                    }
                  />
                )}
              </>
            )}
          </>
        )}
      </StyledBoxContainerList>
    </>
  );
};
