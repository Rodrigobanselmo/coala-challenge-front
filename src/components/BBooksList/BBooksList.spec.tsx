import React from "react";
import { render } from "@testing-library/react";
import { BBooksList } from "./BBooksList";
import { BBooksListProps } from "./types";
import { InfiniteData } from "@tanstack/react-query";
import { IBook } from "@/core/interfaces/IBook";

// Define mock data and props for testing
const mockData: InfiniteData<IBook[]> = {
  pageParams: [],
  pages: [
    [
      {
        title: "Interest Book",
        id: 1,
        createdAt: new Date(),
        updatedAt: new Date(),
        authors: ["Author 1", "Author 2"],
        thumbnail: "/interest-book.png",
      },
      {
        title: "Interest Book4",
        id: 2,
        createdAt: new Date(),
        updatedAt: new Date(),
        authors: ["Author 1", "Author 2"],
        thumbnail: "/interest-book.png",
      },
    ],
    [
      {
        title: "Interest Book 2",
        id: 3,
        createdAt: new Date(),
        updatedAt: new Date(),
        authors: ["Author 1", "Author 2"],
        thumbnail: "/interest-book.png",
      },
    ],
  ],
};

const mockProps: BBooksListProps = {
  data: mockData,
  fetchNextPage: () => {},
  isFetchingNextPage: false,
  isLoading: false,
  onSearchInputChange: () => {},
  searchInputValue: "",
  buttonLabel: "Test Button",
  onClickCardButton: async () => {},
  loadMore: true,
  placeholder: "pesquisar...",
  emptyBookComponent: null,
  emptyBookLable: "Nenhum livro encontrado",
  title: "Book List",
};

describe("BBooksList", () => {
  it("matches the snapshot when loading", () => {
    const { container } = render(
      <BBooksList {...mockProps} isLoading={true} />
    );
    expect(container).toMatchSnapshot();
  });

  it("matches the snapshot with data", () => {
    const { container } = render(<BBooksList {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it("matches the snapshot with no data", () => {
    const { container } = render(
      <BBooksList {...mockProps} data={{ pages: [], pageParams: [] }} />
    );
    expect(container).toMatchSnapshot();
  });

  it("matches the snapshot with empty book component", () => {
    const { container } = render(
      <BBooksList
        {...mockProps}
        emptyBookComponent={<div>Custom Empty Component</div>}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
