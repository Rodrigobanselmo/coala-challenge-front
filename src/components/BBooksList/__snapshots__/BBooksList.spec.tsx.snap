// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BBooksList matches the snapshot when loading 1`] = `
<div>
  <div
    class="MuiBox-root css-1hfnxps"
  >
    <div
      class="MuiBox-root css-1bvhw10"
    >
      <div
        class="MuiBox-root css-70qvj9"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-pwpa85-MuiSvgIcon-root-MuiIcon-root"
          data-testid="AutoStoriesIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m19 1-5 5v11l5-4.5V1zM1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5V6c-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6zm22 13.5V6c-.6-.45-1.25-.75-2-1v13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5v2c1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5v-1.1z"
          />
        </svg>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-1yqseqj-MuiTypography-root"
        >
          Book List
        </h1>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root css-nl8vqx-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          placeholder="pesquisar..."
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <span
    class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
    role="progressbar"
  >
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
    />
    <span
      class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
    />
  </span>
  <div
    class="MuiBox-root css-1qik6aw"
  />
</div>
`;

exports[`BBooksList matches the snapshot with data 1`] = `
<div>
  <div
    class="MuiBox-root css-1hfnxps"
  >
    <div
      class="MuiBox-root css-1bvhw10"
    >
      <div
        class="MuiBox-root css-70qvj9"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-pwpa85-MuiSvgIcon-root-MuiIcon-root"
          data-testid="AutoStoriesIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m19 1-5 5v11l5-4.5V1zM1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5V6c-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6zm22 13.5V6c-.6-.45-1.25-.75-2-1v13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5v2c1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5v-1.1z"
          />
        </svg>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-1yqseqj-MuiTypography-root"
        >
          Book List
        </h1>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root css-nl8vqx-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r1:"
          placeholder="pesquisar..."
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-1qik6aw"
  >
    <div
      class="MuiBox-root css-199gc5h"
    >
      <div
        class="MuiBox-root css-xtnipu"
        data-testid="StyledCard"
      >
        <div
          class="MuiBox-root css-rrl90c"
        >
          <img
            alt="Interest Book"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Finterest-book.png&w=3840&q=75"
            srcset="/_next/image?url=%2Finterest-book.png&w=640&q=75 640w, /_next/image?url=%2Finterest-book.png&w=750&q=75 750w, /_next/image?url=%2Finterest-book.png&w=828&q=75 828w, /_next/image?url=%2Finterest-book.png&w=1080&q=75 1080w, /_next/image?url=%2Finterest-book.png&w=1200&q=75 1200w, /_next/image?url=%2Finterest-book.png&w=1920&q=75 1920w, /_next/image?url=%2Finterest-book.png&w=2048&q=75 2048w, /_next/image?url=%2Finterest-book.png&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
          />
        </div>
        <div
          class="MuiBox-root css-mjw23k"
        >
          <h2
            class="css-yykxhl"
          >
            Interest Book
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 css-17hjpk6-MuiTypography-root"
          >
            Author 1, Author 2
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1g5ile9-MuiButtonBase-root-MuiButton-root"
            is_visible="0"
            tabindex="0"
            type="button"
          >
            Test Button
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root css-xtnipu"
        data-testid="StyledCard"
      >
        <div
          class="MuiBox-root css-rrl90c"
        >
          <img
            alt="Interest Book4"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Finterest-book.png&w=3840&q=75"
            srcset="/_next/image?url=%2Finterest-book.png&w=640&q=75 640w, /_next/image?url=%2Finterest-book.png&w=750&q=75 750w, /_next/image?url=%2Finterest-book.png&w=828&q=75 828w, /_next/image?url=%2Finterest-book.png&w=1080&q=75 1080w, /_next/image?url=%2Finterest-book.png&w=1200&q=75 1200w, /_next/image?url=%2Finterest-book.png&w=1920&q=75 1920w, /_next/image?url=%2Finterest-book.png&w=2048&q=75 2048w, /_next/image?url=%2Finterest-book.png&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
          />
        </div>
        <div
          class="MuiBox-root css-mjw23k"
        >
          <h2
            class="css-yykxhl"
          >
            Interest Book4
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 css-17hjpk6-MuiTypography-root"
          >
            Author 1, Author 2
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1g5ile9-MuiButtonBase-root-MuiButton-root"
            is_visible="0"
            tabindex="0"
            type="button"
          >
            Test Button
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root css-xtnipu"
        data-testid="StyledCard"
      >
        <div
          class="MuiBox-root css-rrl90c"
        >
          <img
            alt="Interest Book 2"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Finterest-book.png&w=3840&q=75"
            srcset="/_next/image?url=%2Finterest-book.png&w=640&q=75 640w, /_next/image?url=%2Finterest-book.png&w=750&q=75 750w, /_next/image?url=%2Finterest-book.png&w=828&q=75 828w, /_next/image?url=%2Finterest-book.png&w=1080&q=75 1080w, /_next/image?url=%2Finterest-book.png&w=1200&q=75 1200w, /_next/image?url=%2Finterest-book.png&w=1920&q=75 1920w, /_next/image?url=%2Finterest-book.png&w=2048&q=75 2048w, /_next/image?url=%2Finterest-book.png&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
          />
        </div>
        <div
          class="MuiBox-root css-mjw23k"
        >
          <h2
            class="css-yykxhl"
          >
            Interest Book 2
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 css-17hjpk6-MuiTypography-root"
          >
            Author 1, Author 2
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1g5ile9-MuiButtonBase-root-MuiButton-root"
            is_visible="0"
            tabindex="0"
            type="button"
          >
            Test Button
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-rqrbu2-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Mostrar mais
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`BBooksList matches the snapshot with empty book component 1`] = `
<div>
  <div
    class="MuiBox-root css-1hfnxps"
  >
    <div
      class="MuiBox-root css-1bvhw10"
    >
      <div
        class="MuiBox-root css-70qvj9"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-pwpa85-MuiSvgIcon-root-MuiIcon-root"
          data-testid="AutoStoriesIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m19 1-5 5v11l5-4.5V1zM1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5V6c-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6zm22 13.5V6c-.6-.45-1.25-.75-2-1v13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5v2c1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5v-1.1z"
          />
        </svg>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-1yqseqj-MuiTypography-root"
        >
          Book List
        </h1>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root css-nl8vqx-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r3:"
          placeholder="pesquisar..."
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-1qik6aw"
  >
    <div
      class="MuiBox-root css-199gc5h"
    >
      <div
        class="MuiBox-root css-xtnipu"
        data-testid="StyledCard"
      >
        <div
          class="MuiBox-root css-rrl90c"
        >
          <img
            alt="Interest Book"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Finterest-book.png&w=3840&q=75"
            srcset="/_next/image?url=%2Finterest-book.png&w=640&q=75 640w, /_next/image?url=%2Finterest-book.png&w=750&q=75 750w, /_next/image?url=%2Finterest-book.png&w=828&q=75 828w, /_next/image?url=%2Finterest-book.png&w=1080&q=75 1080w, /_next/image?url=%2Finterest-book.png&w=1200&q=75 1200w, /_next/image?url=%2Finterest-book.png&w=1920&q=75 1920w, /_next/image?url=%2Finterest-book.png&w=2048&q=75 2048w, /_next/image?url=%2Finterest-book.png&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
          />
        </div>
        <div
          class="MuiBox-root css-mjw23k"
        >
          <h2
            class="css-yykxhl"
          >
            Interest Book
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 css-17hjpk6-MuiTypography-root"
          >
            Author 1, Author 2
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1g5ile9-MuiButtonBase-root-MuiButton-root"
            is_visible="0"
            tabindex="0"
            type="button"
          >
            Test Button
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root css-xtnipu"
        data-testid="StyledCard"
      >
        <div
          class="MuiBox-root css-rrl90c"
        >
          <img
            alt="Interest Book4"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Finterest-book.png&w=3840&q=75"
            srcset="/_next/image?url=%2Finterest-book.png&w=640&q=75 640w, /_next/image?url=%2Finterest-book.png&w=750&q=75 750w, /_next/image?url=%2Finterest-book.png&w=828&q=75 828w, /_next/image?url=%2Finterest-book.png&w=1080&q=75 1080w, /_next/image?url=%2Finterest-book.png&w=1200&q=75 1200w, /_next/image?url=%2Finterest-book.png&w=1920&q=75 1920w, /_next/image?url=%2Finterest-book.png&w=2048&q=75 2048w, /_next/image?url=%2Finterest-book.png&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
          />
        </div>
        <div
          class="MuiBox-root css-mjw23k"
        >
          <h2
            class="css-yykxhl"
          >
            Interest Book4
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 css-17hjpk6-MuiTypography-root"
          >
            Author 1, Author 2
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1g5ile9-MuiButtonBase-root-MuiButton-root"
            is_visible="0"
            tabindex="0"
            type="button"
          >
            Test Button
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root css-xtnipu"
        data-testid="StyledCard"
      >
        <div
          class="MuiBox-root css-rrl90c"
        >
          <img
            alt="Interest Book 2"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Finterest-book.png&w=3840&q=75"
            srcset="/_next/image?url=%2Finterest-book.png&w=640&q=75 640w, /_next/image?url=%2Finterest-book.png&w=750&q=75 750w, /_next/image?url=%2Finterest-book.png&w=828&q=75 828w, /_next/image?url=%2Finterest-book.png&w=1080&q=75 1080w, /_next/image?url=%2Finterest-book.png&w=1200&q=75 1200w, /_next/image?url=%2Finterest-book.png&w=1920&q=75 1920w, /_next/image?url=%2Finterest-book.png&w=2048&q=75 2048w, /_next/image?url=%2Finterest-book.png&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
          />
        </div>
        <div
          class="MuiBox-root css-mjw23k"
        >
          <h2
            class="css-yykxhl"
          >
            Interest Book 2
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body1 css-17hjpk6-MuiTypography-root"
          >
            Author 1, Author 2
          </p>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1g5ile9-MuiButtonBase-root-MuiButton-root"
            is_visible="0"
            tabindex="0"
            type="button"
          >
            Test Button
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-rqrbu2-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Mostrar mais
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`BBooksList matches the snapshot with no data 1`] = `
<div>
  <div
    class="MuiBox-root css-1hfnxps"
  >
    <div
      class="MuiBox-root css-1bvhw10"
    >
      <div
        class="MuiBox-root css-70qvj9"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-pwpa85-MuiSvgIcon-root-MuiIcon-root"
          data-testid="AutoStoriesIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m19 1-5 5v11l5-4.5V1zM1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5V6c-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6zm22 13.5V6c-.6-.45-1.25-.75-2-1v13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5v2c1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5v-1.1z"
          />
        </svg>
        <h1
          class="MuiTypography-root MuiTypography-h1 css-1yqseqj-MuiTypography-root"
        >
          Book List
        </h1>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root css-nl8vqx-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r2:"
          placeholder="pesquisar..."
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-1qik6aw"
  >
    <div
      class="MuiBox-root css-199gc5h"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1bbcyej-MuiPaper-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
      >
        Nenhum livro encontrado
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Adicionar livro
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
